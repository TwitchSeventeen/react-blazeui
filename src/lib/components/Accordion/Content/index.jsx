import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Card from '../../Card';

function Content(props) {
  const {
    children, className, as: Component, dataName, ...other
  } = props;
  const classes = classNames('c-card__item--pane', className);
  return (
    <Component
      {...other}
      className={classes}
      dataName={dataName}
    >
      { children }
    </Component>
  );
}

Content.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Content.defaultProps = {
  children: null,
  className: null,
  dataName: 'accordion-content',
  as: Card.Item,
};

export default Content;
