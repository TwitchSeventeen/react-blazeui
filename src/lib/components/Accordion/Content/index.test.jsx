import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Content from './index';

describe('Content', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Content).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
        <Content>
          lorem ipsum dolor
        </Content>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

