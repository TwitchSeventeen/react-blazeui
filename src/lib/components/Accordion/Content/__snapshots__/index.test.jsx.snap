// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-card__item c-card__item--pane"
    data-name="accordion-content"
  >
    lorem 
    <p>
      lo
    </p>
    lorem ipsum dolor
  </div>
</DocumentFragment>
`;
